WcaOnRails/app/javascript/show-schedule/index.js

```diff
@@ -0,0 +1,243 @@
+  $(idSelector + " .schedule_calendar_container").hide();
+  $(idSelector + " .schedule-calendar-link").removeClass("active");
+  $(idSelector + " .schedule_table_container").show();
+  $(idSelector + " .schedule-table-link").addClass("active");
+}
+
+const onClickCalencarLinkAction = (popoverContentBuilder, locale, startDate, numberOfDays, e) => {
+  e.preventDefault();
+  let venueId = $(e.currentTarget).data("venue");
+  let idSelector = getScheduleElemId(venueId);
+  $(idSelector + " .schedule_table_container").hide();
+  $(idSelector + " .schedule-table-link").removeClass("active");
+  $(idSelector + " .schedule_calendar_container").show();
+  $(idSelector + " .schedule-calendar-link").addClass("active");
+  $calendar = $(getCalendarElemId(venueId));
+  if (!$calendar.hasClass("initialized")) {
+    $calendar.addClass("initialized");
+    let calendarParams = {
+      locale: locale,
+      venueId: venueId,
+      startDate: startDate,
+      numberOfDays: numberOfDays,
+      popoverContentBuilder: popoverContentBuilder,
+    };
+    initFullCalendar($calendar, calendarParams);
+  } else {
+    $calendar.fullCalendar("refetchEvents");
+  }
+}
+
+// Callback when the user click the "all" icon for events
+const onClickAllAction = e => {
+  // The "ALL" button
+  let $allButton = $(e.currentTarget);
+  // The whole filter with all events icons
+  let $filter = $allButton.parent();
+
+  if ($allButton.hasClass("selected")) {
+    // If "ALL" was active, we want to unselect all
+    $filter.find(".cubing-icon.selected").each(function(index, element) {
+      $element = $(element);
+      $element.removeClass("selected");
+      let eventId = $element.data("event");
+      // Table view action
+      $(".schedule-table .event-" + eventId).removeClass("event-selected");
+    });
+    $allButton.removeClass("selected");
+  } else {
+    $filter.find(".cubing-icon:not(.selected)").each(function(index, element) {
+      $element = $(element);
+      $element.addClass("selected");
+      let eventId = $element.data("event");
+      // Table view action
+      $(".schedule-table .event-" + eventId).addClass("event-selected");
+    });
+    $allButton.addClass("selected");
+  }
+  let venueId = $filter.data("venue");
+  // Calendar action
+  maybeRefreshEvents(venueId);
+}
+
+// Callback when the user click an event icon (actually updates *all* the filters in all venues)
+const onClickEventIconAction = e => {
+  // The event icon
+  let $elem = $(e.currentTarget);
+  let $filter = $elem.parent();
+  // Event id for this event icon
+  let eventId = $elem.data("event");
+
+  // Toggle the event visibility
+  if ($elem.hasClass("selected")) {
+    $elem.removeClass("selected");
+    // Table action
+    $(".schedule-table .event-" + eventId).removeClass("event-selected");
+  } else {
+    $elem.addClass("selected");
+    // Table action
+    $(".schedule-table .event-" + eventId).addClass("event-selected");
+  }
```

Comment:
Thanks, I didn't know this function!

Reply:
Actually I changed it to
```js
  $elem.toggleClass("selected");
  $(".schedule-table .event-" + eventId).toggleClass("event-selected");
```
Since here we don't want to enforce a value but only to toggle it :)